### ISSUE TITLE
textDocument/signatureHelp received after LSP exit

### ISSUE THREAD

**Issue by kting28** [ray-x/lsp_signature.nvim#258]
With lsp_signature enabled with clangd, I frequently get these error messages printed on the console:

```
RPC[Error] code_name = UnknownErrorCode, message = “Failed to parse includes”
```
After some investigations it seems to be related to LSP existing before the signature query reply is sent back. Sample steps:

1. Open a file with LSP is enabled and wait for parsing to start 
2. Exit the editor
3. Some `RPC[Error]` messages are printed on the console. From the lsp.log, these seems to be received (or sent?) after the LSP has exited. 

For example in the log below, you can see LSP has finished then next line the RPC error is received and printed on stderr
```
[ERROR][2023-02-20 20:19:36] .../vim/lsp/rpc.lua:733	"rpc"	"clangd"	"stderr"	'V[20:19:36.934] <<< {"jsonrpc":"2.0","method":"exit"}\n\nI[20:19:36.934] <-- exit\nI[20:19:36.934] LSP finished, exiting with status 0\n'
[ERROR][2023-02-20 20:19:36] .../vim/lsp/rpc.lua:733	"rpc"	"clangd"	"stderr"	'I[20:19:36.934] --> reply:textDocument/signatureHelp(4) 3165 ms, error: Failed to parse includes\nV[20:19:36.934] >>> {"error":{"code":-32001,"message":"Failed to parse includes"},"id":4,"jsonrpc":"2.0"}\n\n'
[DEBUG][2023-02-20 20:19:36] .../vim/lsp/rpc.lua:388	"rpc.receive"	{  error = {    code = -32001,    message = "Failed to parse includes"  },  id = 4,  jsonrpc = "2.0"}
[ERROR][2023-02-20 20:19:36] .../vim/lsp/rpc.lua:733	"rpc"	"clangd"	"stderr"	"V[20:19:36.936] ASTWorker rebuilding evicted AST to run DocumentSymbols: <--redacted path to file--> version 0\n"
```
This is a minor issue as the message are only printed on exit but it's quite annoying.  Anyway  to supress?

**Comment by kting28** [comment]
btw I don't seem to see the same issue with other signature help plugins such as
[cmp-nvim-lsp-signature-help](https://github.com/hrsh7th/cmp-nvim-lsp-signature-help). Is this related to the "async" signature requests this plugin is design to do? I.e. LSP is stopped while there are outstanding queries?

**Comment by ray-x** [comment]
Do you mean when you quit nvim you saw `RPC[Error] code_name = UnknownErrorCode, message = “Failed to parse includes”` ?

**Comment by kting28** [comment]
> Do you mean when you quit nvim you saw `RPC[Error] code_name = UnknownErrorCode, message = “Failed to parse includes”` ?

Yup. It's not consistent but for larger files I see multiple lines of these. Does not happen if I waited long enough after opening a file; probably after all queries are made?

**Comment by ray-x** [comment]
Ok, I think it is related to the async request I sent. But I have not seen the same issue so far. Might be also because the clangd is slow or your project is large.
How you exit your nvim? Do you have a keybind or you do `:wq` from cmdline? More specifically, are you exit `Insert` mode before you exit?

**Comment by kting28** [comment]
Yea the project is plenty large. It takes 10 seconds to index from scratch
and a few more for diagnostics to be loaded in plugins like Trouble.

I have only tried  :q , not from insert, was just inspecting the files. If
I stay in the file for a while the chance of seeing these are less.
